partiql-parser/src/main/antlr/PartiQL.g4

```diff
@@ -812,12 +812,19 @@ type
     : datatype=(
         NULL | BOOL | BOOLEAN | SMALLINT | INTEGER2 | INT2 | INTEGER | INT | INTEGER4 | INT4
         | INTEGER8 | INT8 | BIGINT | REAL | CHAR | CHARACTER | MISSING
-        | STRING | SYMBOL | BLOB | CLOB | DATE | STRUCT | TUPLE | LIST | SEXP | BAG | ANY
+        | STRING | SYMBOL | BLOB | CLOB | DATE | SEXP | BAG | ANY
       )                                                                                                                # TypeAtomic
+    | datatype=(STRUCT|TUPLE|LIST|ARRAY)                                                                               # TypeComplexUnparameterized
     | datatype=DOUBLE PRECISION                                                                                        # TypeAtomic
     | datatype=(CHARACTER|CHAR|FLOAT|VARCHAR) ( PAREN_LEFT arg0=LITERAL_INTEGER PAREN_RIGHT )?                         # TypeArgSingle
     | CHARACTER VARYING ( PAREN_LEFT arg0=LITERAL_INTEGER PAREN_RIGHT )?                                               # TypeVarChar
     | datatype=(DECIMAL|DEC|NUMERIC) ( PAREN_LEFT arg0=LITERAL_INTEGER ( COMMA arg1=LITERAL_INTEGER )? PAREN_RIGHT )?  # TypeArgDouble
     | datatype=(TIME|TIMESTAMP) ( PAREN_LEFT precision=LITERAL_INTEGER PAREN_RIGHT )? (WITH TIME ZONE)?                # TypeTimeZone
+    | datatype=(STRUCT|TUPLE) (ANGLE_LEFT structAttr ( COMMA structAttr )* ANGLE_RIGHT)                                # TypeStruct
```

Comment:
Since this is going to the `1.0`, do we want to flesh out the open/close and other struct constraints in the syntax now; if not, how can we avoid backward incompatibility when we introduce the syntax for those?

Reply:
In my mental model, I think that using SQL's `CREATE TABLE` get us a bag of closed struct. For creating shapes that is unique to PartiQL, a bag of opened struct, a bag of integers, etc. we can consider vend out a PartiQL specific syntax (i.e., CREATE SHAPE ?) for those.

Reply:
Just like other deviations from Hive DDL, we may have to support such use-cases within the same syntax; let's discuss further about this.

Reply:
Adding the summary of our discussion in an example here for posterity:

```SQL
CREATE TABLE [OPEN] Foo (
   s STRUCT<a: INT, b:STRING> OPEN,
)
```